helpers/auth/admin_roles.py

```diff
@@ -30,6 +33,24 @@ def create_office(self, location_id):
         if location.name != admin_details.location:
             raise GraphQLError("You are not authorized to make changes in " + location.name)  # noqa: E501
 
+    def check_office_location_create_wing(self, floor_id):
+        get_floor = FloorModel.query.filter_by(id=floor_id).first()
+        get_block = Block.query.filter_by(id=get_floor.block_id).first()  # noqa: E501
+        get_office = Office.query.filter_by(id=get_block.office_id).first()  # noqa: E501
+        get_location = Location.query.filter_by(id=get_office.location_id).first()  # noqa: E501
+        if get_location.name.lower() != 'lagos':
+            raise GraphQLError("This action is restricted to Lagos Office only")  # noqa: E501
+
+    def create_update_delete_wing(self):
+        admin_details = get_user_from_db()
+        if admin_details.location.lower() != 'lagos':
+            raise GraphQLError("This action is restricted to Lagos Office admin")  # noqa: E501
+
+    def check_wing(self, name):
+        existing_wing = Wing.query.filter_by(name=name.title()).first()
+        if existing_wing:
+            raise GraphQLError("You can only create " + name + " wing once")  # noqa: E501
```

Comment:
It would make more sense to say `wing already exists` for this error. it would also be better to use an existing helper `saveContextManager` which already implements database conflicts. check helpers/error_handler

Reply:
Thanks, @shemogumbe, I have implemented the requested changes